examples/nidcpower_source_dc_voltage/teststand_fixture.py

```diff
@@ -39,48 +44,47 @@ def create_nidcpower_sessions(sequence_context: Any) -> None:
             (Dynamically typed.)
     """
     with GrpcChannelPoolHelper() as grpc_channel_pool:
-        session_management_client = nims.session_management.Client(
-            grpc_channel=grpc_channel_pool.session_management_channel
-        )
-
         teststand_support = TestStandSupport(sequence_context)
         pin_map_id = teststand_support.get_active_pin_map_id()
-        pin_map_context = nims.session_management.PinMapContext(pin_map_id=pin_map_id, sites=None)
-        grpc_device_channel = grpc_channel_pool.get_grpc_device_channel(
-            nidcpower.GRPC_SERVICE_INTERFACE_NAME
+        pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=None)
+
+        discovery_client = DiscoveryClient(grpc_channel_pool=grpc_channel_pool)
+        session_management_client = SessionManagementClient(
+            discovery_client=discovery_client, grpc_channel_pool=grpc_channel_pool
         )
         with session_management_client.reserve_sessions(
-            context=pin_map_context,
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
+            pin_map_context, instrument_type_id=INSTRUMENT_TYPE_NI_DCPOWER
         ) as reservation:
-            for session_info in reservation.session_info:
-                # Leave session open
-                _ = create_session(
-                    session_info,
-                    grpc_device_channel,
-                    initialization_behavior=nidcpower.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
-                )
+            # Initialize new sessions on the server, but do not close them.
+            # (Deliberately leak them.)
+            context_manager = reservation.create_nidcpower_sessions(
+                initialization_behavior=SessionInitializationBehavior.INITIALIZE_SERVER_SESSION
+            )
+            _ = context_manager.__enter__()
```

Comment:
Do we want something less subtle/advanced here?

Ideas:
1. Add an `INITIALIZE_AND_DETACH` enum value to measurementlink-python and move this logic into measurementlink-python, like we are doing for VISA
2. Add an `INITIALIZE_AND_DETACH` enum value to nimi-python and nidaqmx-python, like we are doing for VISA
3. Add a detach_session() method to the measurementlink-python `SessionInformation` object
4. Add a detach() method to nimi-python and nidaqmx-python

Reply:
5. Add `reservation.detach_sessions()`

Reply:
I have been thinking about this from the LV perspective as well, and I'm leaning more towards having a single enum that specifies both init and close behavior at the same time. It would also have an "Auto" option if that makes sense, where the auto option would calculate initialize behavior at run time and apply symmetrical cleanup behavior.

I think this makes sense because:
1. It makes sense for languages where cleanup can be provided by exit from a scope or context. This also applies to LV when the top level VI goes idle for refnums that have a cleanup proc.
2. Lowest common denominator. The VISA driver API is planning to provide an enum like this. If we don't offer a similar API to our wrapper APIs, then how will it integrate with driver APIs like VISA?

Reply:
I'm not opposed to pushing more of this capability directly into the driver APIs. However, I think we can also use the MeasurementLink layer to provide a more consistent API/experience across drivers as well.

Reply:
The Python session management API defines a driver-independent version of this enum: https://github.com/ni/measurementlink-python/blob/main/ni_measurementlink_service/session_management/_types.py#L289

and maps it to each driver-specific enum:
https://github.com/ni/measurementlink-python/blob/main/ni_measurementlink_service/_drivers/_nidcpower.py#L18

Proposal:
- Change the MeasurementLink version of the enum to have 5 options (AUTO, INIT_AND_CLOSE, INIT_AND_DETACH, ATTACH_AND_CLOSE, ATTACH_AND_DETACH)
- Map these 5 options to the 3 options supported by nimi-python and nidaqmx-python
- Implement the 2 "_AND_CLOSE" options by explicitly calling close

Reply:
> I'm not opposed to pushing more of this capability directly into the driver APIs. However, I think we can also use the MeasurementLink layer to provide a more consistent API/experience across drivers as well.

I agree.

Also, we can do it in the MeasurementLink layer now without a driver API release.

Reply:
That's more or less what I was thinking for LV as well. It would mean bubbling that input up to all the driver specific Create Session APIs. The parameter would default to AUTO so it wouldn't have to be explicitly set as part of the measurement logic. FYI, @pbirkhol-ni , I think you had some opinions about this. If you have objections or concerns to this approach, now is a good time to voice them.

Reply:
Ok, I added AB#2552420

Reply:
> refer to VISA for exact naming

Do you have a link for this?

Can you explain each of the 5 options in terms of what is returned (it returns a ContextManager right?) and what state that context manager is in?

For example, does INIT_AND_CLOSE mean that we will close the session(s) when the ContextManager we return to you is exited? And INIT_AND_DETACH means that the session(s) will not be closed at all by us?

Reply:
> If you have objections or concerns to this approach, now is a good time to voice them.

I don't have any major concerns. The enum option is better than any alternatives I have heard. @jasonmreding, can you add a user story for LabVIEW. This change would be hard, but will involve touching a lot of VIs, updating tests, and writing new tests.

Reply:
> > refer to VISA for exact naming
> 
> Do you have a link for this?

https://dev.azure.com/ni/DevCentral/_git/ni-central?path=%2Fdocs%2Fspecs%2Fvisa%2Fnivisa_grpc_client_support_for_labview%2Fvisa_grpc_client_support_for_labview_api.md&version=GBmain&_a=preview

> Can you explain each of the 5 options in terms of what is returned (it returns a ContextManager right?) and what state that context manager is in?
> 
> For example, does INIT_AND_CLOSE mean that we will close the session(s) when the ContextManager we return to you is exited? And INIT_AND_DETACH means that the session(s) will not be closed at all by us?

The context manager that we return is implemented as a generator in [_create_session(s)_core](https://github.com/ni/measurementlink-python/blob/main/ni_measurementlink_service/session_management/_reservation.py#L285).

I think we can implement this by updating closing_session(). This currently chooses between using the session as a context manager or returning contextlib.closing() to call close().

The updated behavior will be something like this (assuming the session is both a context manager and closable, which we can make a requirement):

- AUTO: use the session's context manager
- INIT_AND_CLOSE: use contextlib.closing
- INIT_AND_DETACH: use contextlib.nullcontext
- ATTACH_AND_CLOSE: use contextlib.closing
- ATTACH_AND_DETACH: use contextlib.nullcontext

Reply:
> can you add a user story for LabVIEW

This is what AB#2552373 is intended to capture. The API changes will impact both code modules and measurements so you're welcome to move ownership of the user story if you think it is better tracked somewhere else.